src/features/More/components/categoryItem.tsx

```diff
@@ -0,0 +1,67 @@
+import React from "react";
+import { StyleSheet, Text, TouchableOpacity, View } from "react-native";
+import ArrowIcon from "../../../assets/icons/ArrowIcon";
+import Colors from "../../../constants/Colors";
+import Category from "../more.model";
+
+interface Props {
+  itemData: { item: Category; index: number };
+  indexOfPressedCategory: number | null;
```

Comment:
I think it will be more readable if you would have a prop called selected and the logic will be in the parent component